RecoParticleFlow/PFProducer/plugins/MLPFProducerSonic.cc

```diff
@@ -0,0 +1,199 @@
+#include <sstream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <cmath>
+#include <map>
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlock.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementSuperCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementGsfTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementBrem.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElement.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicEDProducer.h"
+#include "HeterogeneousCore/SonicTriton/interface/TritonClient.h"
+#include "RecoParticleFlow/PFProducer/interface/MLPFModel.h"
+
+#include "TLorentzVector.h"
+
+// git clone https://github.com/jpata/mlpf-models.git
+// mv mlpf-models/triton/models ./
+// singularity exec --nv docker://nvcr.io/nvidia/tritonserver:20.10-py3 tritonserver --model-repository models --log-verbose=1 --log-error=1 --log-info=1
+//
+// process.mlpfproducergpu = cms.EDProducer("MLPFProducerSonicTriton",
+//     Client = cms.PSet(
+//         mode = cms.string("Sync"),
+//         batchSize = cms.untracked.uint32(1),
+//         address = cms.untracked.string("127.0.0.1"),
+//         port = cms.untracked.uint32(8001),
+//         timeout = cms.untracked.uint32(60),
+//         modelName = cms.string("mlpf"),
+//         modelVersion = cms.string("1"),
+//         verbose = cms.untracked.bool(False),
+//         allowedTries = cms.untracked.uint32(0),
+//     )
+// )
+
+class MLPFProducerSonicTriton : public SonicEDProducer<TritonClient> {
+public:
+  //needed because base class has dependent scope
+  explicit MLPFProducerSonicTriton(edm::ParameterSet const& cfg)
+      : SonicEDProducer<TritonClient>(cfg),
+        pfCandidatesToken_{this->template produces<reco::PFCandidateCollection>()},
```

Comment:
`template ` not needed anymore before member functions
in general, `this->` should not be needed either (in most cases - try removing it anywhere and check compilation)